tensorflow/core/transforms/remapper/pass.cc

```diff
@@ -91,18 +96,220 @@ class MatchMulSigmoid : public RewritePattern {
   OperationName sigmoid_name_;
 };
 
+namespace {
+class RemapperPatternBase : public RewritePattern {
+ public:
+  RemapperPatternBase(StringRef opName, OpPropertyHelper &helper,
+                      PatternBenefit benefit = PatternBenefit(1))
+      : RewritePattern(opName, benefit, helper.getDialect()->getContext()),
+        helper_(helper) {}
+  RemapperPatternBase(MatchAnyOpTypeTag tag, OpPropertyHelper &helper,
+                      PatternBenefit benefit = PatternBenefit(1))
+      : RewritePattern(tag, benefit, helper.getDialect()->getContext()),
+        helper_(helper) {}
+
+ protected:
+  OpPropertyHelper helper_;
+};
+}  // namespace
+
+static std::unique_ptr<OperationState> GetContractionBiasAddOpState(
+    OpBuilder &builder, const OpPropertyHelper &helper,
+    Operation *contraction_op, Operation *bias_add_op) {
+  // Fused op name dependes on original contraction op name.
+  std::string fused_op_name;
+  if (helper.getDialect()->IsConv2D(contraction_op)) {
+    fused_op_name = "tfg._FusedConv2D";
+  } else if (helper.getDialect()->IsMatMul(contraction_op)) {
+    fused_op_name = "tfg._FusedMatMul";
+  } else if (helper.getDialect()->IsDepthwiseConv2dNative(contraction_op)) {
+    fused_op_name = "tfg._FusedDepthwiseConv2dNative";
+  } else if (helper.getDialect()->IsConv3D(contraction_op)) {
+    fused_op_name = "tfg._FusedConv3D";
+  } else {
+    return nullptr;
+  }
+
+  SmallVector<Location> fused_locs{contraction_op->getLoc(),
```

Comment:
@ChiaHungDuan I am not sure what should be set as a location of a fused. Is the current approach looks good to you?

Reply:
LGTM